### DISCUSSION TITLE
lightSleep not very light.

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#9569] [General]
According to several sources, while in "lightsleep" an ESP8266 should draw at most around 2ma, and still maintain the GPIO states. Testing did not confirm the low current; while in lightsleep the unit continued to draw it's normal running current of approximately 78ma.

The test was performed with the following instruction:
machine.lightsleep(5000) # 5000ms for a 5 second sleep.

The ESP8266 did indeed delay for 5 seconds and maintained the state of the GPIO pin of interest (GPIO4) at a low (drawing about 0.13ma).

Can anyone suggest why the current draw during lightsleep is much higher than expected?

**Comment by robert-hh** [comment]
Was WiFi enabled? WiFi must be switched off for lightsleep

**Reply by mmliam** [reply]
Ah, I didn't disable the wifi; thought that would have been part of the lightsleep mode.
Thanks Robert.

**Reply by mmliam** [reply]
Hi Robert,
I deactivated the wifi [eNet.active(false)], but that didn't make any difference. Is there some other wifi power instruction?
I've seen references to MODEM_SLEEP, but not in micropython. There is also machine.idle() that might be useful as long as the state of the GPIO pins is maintained.

Is it possible that to get the full power saving benefit of "lightsleep" you can't specify an argument, you need to use the WDT to wake-up the unit?

**Reply by mmliam** [reply]
Robert,
According to this table, taken from an Espressif document; wifi is supposed to be off in light-sleep.
![Sleep-Mode Table](https://user-images.githubusercontent.com/96342918/194954032-5e5daf3a-2670-494e-ad94-736e39601138.JPG)

**Reply by mmliam** [reply]
Hi Robert,
BTW I'm not using wifi per say, I'm using ESPNow, although it uses the wifi radio, it's not wifi. However, I created an object for the STA & AP and set them both to active-False. After a transmit/receive cycle with an ESPNow receiver I de-initialized the ESPNow object, but during machine.lightsleep(5000) the ESP8266 was still drawing it's normal running current of 78ma. Not sure what's left to try to get the current draw down to the specified low current.

I guess the question is, why does every website, including Espressif say Light-sleep will shutoff the wifi and draw low current when it doesn't; is this just a problem with the implementation in microPython? If it is this should be the developers first order of business to fix. Deep-sleep does shut the wifi down, and I would use it, if it were not for the fact that I need the state of the pins (or at least one pin) to maintain it's state during sleep. 

From some old posts I've read, it seems even DP George hasn't found a solution as far as Light-sleep goes for microPython, even though it works in Arduino. I still don't understand, if George was able to get the wifi to shutdown during Deep-sleep, why he can't implement the relevant code for Light-sleep.

**Reply by mmliam** [reply]
Hi Robert,
At long last I had some success; I've managed to reduce the running current from 80ma, down to 20ma, and then 17ma during Light-sleep; while it's not 2ma, it's better than the constant 80ma. I'm thinking that the device may have connected to my router and by including a disconnect before active(False) may be the reason for the lower current draw. It's still not totally disabling the wifi radio, but from the past posts I've seen no one including DP George have had success with Light-sleep totally shutting down the wifi.
The current still spikes up, but now only during a brief transmission.

**Comment by davefes** [comment]
... and there is no USB to serial chip on the board?

**Reply by mmliam** [reply]
There's no onboard USB.
